v2/sourcedb-to-spanner/src/main/java/com/google/cloud/teleport/v2/source/reader/io/row/SerializableGenericRecord.java

```diff
@@ -0,0 +1,127 @@
+/*
+ * Copyright (C) 2024 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package com.google.cloud.teleport.v2.source.reader.io.row;
+
+import java.io.IOException;
+import java.io.ObjectStreamException;
+import java.io.Serializable;
+import java.util.Objects;
+import org.apache.avro.Schema;
+import org.apache.avro.generic.GenericDatumReader;
+import org.apache.avro.generic.GenericDatumWriter;
+import org.apache.avro.generic.GenericRecord;
+import org.apache.avro.io.BinaryDecoder;
+import org.apache.avro.io.BinaryEncoder;
+import org.apache.avro.io.DecoderFactory;
+import org.apache.avro.io.EncoderFactory;
+
+/**
+ * Wrap {@link GenericRecord} into a serializable type.
+ *
+ * <p>Unlike, {@link org.apache.avro.specific.SpecificRecord}, {@code GenericRecord}s do not have a
+ * schema at compile time. Hence, a genericRecord needs a custom serde that encodes both the schema
+ * and the data.
+ *
+ * <p><b>Note:</b>
+ *
+ * <ol>
+ *   <li>Based on benchmarking results, this can be further enhanced to have a singleton schema
+ *       registry and encode only the hash of the schema in the serialization.
+ *   <li>This can't be made as an {@link com.google.auto.value.AutoValue} class, as they are not
+ *       mutable and do not support custom serializations.
+ * </ol>
+ */
```

Comment:
We do use builder to build the GenericRecord. That's the part of building `SourceRow`

GenericRecord does not implement Serializable. `SpecificRecord` does, but we cant use that since it needs a fixed schema.

This class is just to implement the serializable interface on GenericRecord.

Reply:
Why can't we use `GenericData.Record`? I see some code references where it is the output of a DoFn - 

https://github.com/GoogleCloudPlatform/DataflowTemplates/blob/69f26e86e095a733316f63da1af30bf77fb4f742/v2/common/src/main/java/com/google/cloud/teleport/v2/transforms/CsvConverters.java#L644


https://github.com/GoogleCloudPlatform/DataflowTemplates/blob/69f26e86e095a733316f63da1af30bf77fb4f742/v2/common/src/main/java/com/google/cloud/teleport/v2/transforms/CsvConverters.java#L682

Reply:
Have taken a TODO to explore a coder. Resolving for the scope of this PR.